File: Freedom/微信/通讯录/WXPublicServerViewController.m
//  TLPublicServerViewController.m
//  Freedom
// Created by Super
#import "WXPublicServerViewController.h"
#import "WXPublicServerSearchViewController.h"
#import "WXSearchController.h"
@interface WXPublicServerViewController () <UISearchBarDelegate>
@property (nonatomic, strong) WXSearchController *searchController;
@property (nonatomic, strong) WXPublicServerSearchViewController *searchVC;
@end
@implementation WXPublicServerViewController
- (void)viewDidLoad{
    [super viewDidLoad];
    [self.navigationItem setTitle:@"公众号"];
    [self.view setBackgroundColor:[UIColor whiteColor]];
    
    UIBarButtonItem *rightBarButton = [[UIBarButtonItem alloc] initWithImage:[UIImage imageNamed:@"nav_add"] style:UIBarButtonItemStylePlain target:self action:@selector(rightBarButtonDown:)];
    [self.navigationItem setRightBarButtonItem:rightBarButton];
    
    [self.tableView setTableHeaderView:self.searchController.searchBar];
}
#pragma mark - Event Response -
- (void)rightBarButtonDown:(UIBarButtonItem *)sender{
}
#pragma mark - Delegate -
//MARK: UISearchBarDelegate
- (void)searchBarTextDidBeginEditing:(UISearchBar *)searchBar{
    
}
- (void)searchBarTextDidEndEditing:(UISearchBar *)searchBar{
}
#pragma mark - Getter -
- (WXSearchController *)searchController{
    if (_searchController == nil) {
        _searchController = [[WXSearchController alloc] initWithSearchResultsController:self.searchVC];
        [_searchController setSearchResultsUpdater:self.searchVC];
        [_searchController.searchBar setPlaceholder:@"搜索"];
        [_searchController.searchBar setDelegate:self];
    }
    return _searchController;
}
- (WXPublicServerSearchViewController *)searchVC{
    if (_searchVC == nil) {
        _searchVC = [[WXPublicServerSearchViewController alloc] init];
    }
    return _searchVC;
}
@end

File: Freedom/微信/通讯录/WXPublicServerViewController.swift
//
//  WXPublicServerViewController.swift
//  Freedom
import XExtension
import Foundation
class WXPublicServerViewController: BaseTableViewController, UISearchBarDelegate {
    lazy var searchController: WXSearchController = {
        let searchController = WXSearchController(searchResultsController: searchVC)
        searchController.searchResultsUpdater = searchVC
        searchController.searchBar.placeholder = "搜索"
        searchController.searchBar.delegate = self
        return searchController
    }()
    var searchVC = WXPublicServerSearchViewController()
    override func viewDidLoad() {
        super.viewDidLoad()
        navigationItem.title = "公众号"
        view.backgroundColor = UIColor.whitex
        let rightBarButton = UIBarButtonItem(image: Image.add.image, style: .plain, target: self, action: #selector(WXPublicServerViewController.rightBarButtonDown(_:)))
        navigationItem.rightBarButtonItem = rightBarButton
        tableView.tableHeaderView = searchController.searchBar
    }
    @objc func rightBarButtonDown(_ sender: UIBarButtonItem) {
    }
    func searchBarTextDidBeginEditing(_ searchBar: UISearchBar) {
    }
    func searchBarTextDidEndEditing(_ searchBar: UISearchBar) {
    }
}
